src/components/Message/__tests__/utils.test.js

```diff
@@ -324,4 +326,54 @@ describe('Message utils', () => {
       expect(getNonImageAttachments(message)).toStrictEqual([pdf]);
     });
   });
+
+  describe('getReadbyTooltipText', () => {
+    let client;
+
+    beforeAll(async () => {
+      client = await getTestClientWithUser(alice);
+    });
+    it('ignores the client user', () => {
+      const result = getReadByTooltipText([client.user], mockTranslatorFunction, client);
+      expect(result).toStrictEqual('');
+    });
+    it('returns a single user if only one user in array', () => {
+      const result = getReadByTooltipText([bob], mockTranslatorFunction, client);
+      expect(result).toStrictEqual(`${bob.name} `);
+    });
+    it('returns "1 and 2" if read by two users', () => {
+      const users = [generateUser({ name: '1' }), generateUser({ name: '2' })];
+      const result = getReadByTooltipText(users, mockTranslatorFunction, client);
+      expect(result).toStrictEqual(`1 and 2`);
+    });
+    it('returns "1, 2, and 3" if read by three users', () => {
+      const users = [
+        generateUser({ name: '1' }),
+        generateUser({ name: '2' }),
+        generateUser({ name: '3' }),
+      ];
+      const result = getReadByTooltipText(users, mockTranslatorFunction, client);
+      expect(result).toStrictEqual(`1, 2, and 3`);
+    });
+    it('returns "1, 2, 3, 4, 5, and 5 more if read by ten users', () => {
+      const users = [...Array(10).keys()].map((n) => generateUser({ name: (n + 1).toString() }));
+      const result = getReadByTooltipText(users, mockTranslatorFunction, client);
+      expect(result).toStrictEqual(`1, 2, 3, 4, 5, and 5 more`);
+    });
+    it('overrides user format with readByToolTipFormat', () => {
```

Comment:
```suggestion
    it('overrides user format with tooltipUserNameMapper', () => {
```

Reply:
With my suggestions, the mapper function isn't optional anymore and it has to be provided as an argument to `getReadByTooltiptText` in these tests as well. Can you please take care of that?

Reply:
Why would you prefer that to be a required parameter rather than optional? Sorry for the late reply, last week was hectic for me.

Reply:
Also, where should I add the docs for the MessageStatus?  Should I create a new mdx file?   What's the difference between `docusaurus/docs/React/customization/message-ui.mdx` vs `docusaurus/docs/React/message-components/message-ui.mdx`?

Reply:
Hey @jfacoustic, no problem at all.

> Why would you prefer that to be a required parameter rather than optional?


`getReadByTooltipText` is internal API. If we have a stricter contract on it, we reduce the probability of future bugs. On the other hand, the MessageStatus API is public and there we are totally ok to have an optional parameter, as long as we are providing a default value for it.
For the tests, we can also use the default implementation. It can be exported from `utils.ts`.

Reply:
> Also, where should I add the docs for the MessageStatus? Should I create a new mdx file? What's the difference between `docusaurus/docs/React/customization/message-ui.mdx` vs `docusaurus/docs/React/message-components/message-ui.mdx`?

Please add the docs [here](https://github.com/GetStream/stream-chat-react/blob/develop/docusaurus/docs/React/message-components/ui-components.mdx#messagestatus-props).

`customization/message-ui.mdx` serves as an example of how a custom message can be implemented. The API docs live in the other file.

Reply:
@oliverlaz Added the docs and fixed the code comments!  Let me know if it needs anything else.